src/views/Home.jsx

```diff
@@ -1,22 +1,57 @@
 import './Home.css';
 import { useNavigate } from 'react-router-dom';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+import { findToken } from '../api/firebase';
 
-export function Home({ handleClick, listToken }) {
+export function Home({ handleClick, listToken, setListToken }) {
 	const navigateTo = useNavigate();
+	const [joinListName, setJoinListName] = useState('');
+	const [error, setError] = useState(false);
 
 	useEffect(() => {
 		if (listToken) {
 			navigateTo('/list');
 		} else return;
 	}, [listToken, navigateTo]);
 
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setError(false);
+		const querySnapshot = await findToken(joinListName);
+		if (querySnapshot.size >= 1) {
+			setListToken(joinListName);
+			navigateTo('/list');
+		} else if (querySnapshot.empty) {
+			setError(true);
+		}
+	};
+
 	return (
 		<div className="Home">
-			<p>
-				Hello from the home (<code>/</code>) page!
-			</p>
 			<button onClick={handleClick}>Create New List</button>
+
+			<div>
+				<form onSubmit={handleSubmit}>
+					<div className="list-name">
+						<label htmlFor="list-name">
+							List Name:{' '}
```

Comment:
Great work with the form and the default submit events!

Reply:
Refactor: use CSS margin instead of a blank space to put space between the label and input.